test/common/inspector-helper.js

```diff
@@ -412,6 +412,43 @@ function readMainScriptSource() {
   return fs.readFileSync(_MAINSCRIPT, 'utf8');
 }
 
+function onResolvedOrRejected(promise, callback) {
+  return promise.then((result) => {
+    callback();
+    return result;
+  }, (error) => {
+    callback();
+    throw error;
+  });
+}
+
+function timeoutPromise(error, timeoutMs) {
+  let clearCallback = null;
+  let done = false;
+  const promise = onResolvedOrRejected(new Promise((resolve, reject) => {
```

Comment:
Actually, this code for `timeoutPromise` is a little convoluted IMO - any reason we can't do.

```js
const promisify = require('util').promisify;
const delay = promisify(setTimeout);
const timeoutPromise = async (error, timeoutMs, cancelToken) => {
  await delay(ms);
  if(cancelToken.cancelled) { 
    return;
  }
  throw error;
};
```

Turns fires to:

```js
function fires(promise, error, timeoutMs) {
  var token = {};
  return Promise.race([
     onResolvedOrRejected(promise, () => token.cancelled = true),
     timeoutPromise(error, timeoutMs, token)
   ]);
 }
```

If we want to preserve the `onResolvedOnRejected` behavior (which I'm not sure I understand) we can just use `Promise.prototype.finally` which is in V8 6.3

Reply:
No issues with further refactoring, but I'd prefer to do that in a separate PR.